### ISSUE TITLE
bug in --auto-correct ?

### ISSUE THREAD

**Issue by daddygongon** [bbatsov/rubocop#4644]
Dear sirs
I am really a novice of rubocop.  When I try auto-correct a part of code
``` ruby
        3.times{|i|
          if (pos_i[i]==1.0 or pos_j[i]==1.0)
            bias = 1.0
          else
            bias = 0.0
          end
#          bias = (pos_i[i]==1.0 or pos_j[i]==1.0) ? 1.0 : 0.0
        }
```
by the command,
```
rubocop --auto-correct test.rb
```
the rubocop did
``` ruby
3.times { |i|
          bias = if pos_i[i] == (.0) || pos_j[i] == (.0)
            1.0
          else
            0.0
                 end
          #          bias = (pos_i[i]==1.0 or pos_j[i]==1.0) ? 1.0 : 0.0
        }
```
The values in the if-condition were incorrectly translated.  Is this OK?
I know I was nuts on the comparison with the floating point... a very old code I wrote...

**Comment by mikegee** [comment]
> Is this OK?

No, it is broken.

I confirmed this behavior with the latest `master`.

**Comment by daddygongon** [comment]
Hello,
Thanks for the fast responce.  Another typical bad example ...
cat tmp_before_auto_correct.rb tmp_after_auto_correct.rb
j_pos[1] > - @rad && j_pos[1] < + @rad
j_pos[1]>y-@rad and j_pos[1]<y+@rad

This is good one.
j_pos[1]>y-@rad
j_pos[1] > y - @rad

**Comment by daddygongon** [comment]
Sorry, the  order of files are opposite.
cat tmp_before_auto_correct.rb tmp_after_auto_correct.rb
j_pos[1]>y-@RaD and j_pos[1]<y+@RaD
j_pos[1] > - @RaD && j_pos[1] < + @RaD

This is good one.
cat tmp_before_auto_correct.rb tmp_after_auto_correct.rb
j_pos[1]>y-@RaD
j_pos[1] > y - @RaD

**Comment by jonas054** [comment]
The current `master` auto-corrects to
```ruby
3.times do |i|
  bias = if (pos_i[i] == 1.0) || (pos_j[i] == 1.0)
           1.0
         else
           0.0
         end
  #          bias = (pos_i[i]==1.0 or pos_j[i]==1.0) ? 1.0 : 0.0
end

(j_pos[1] > y - @RaD) && (j_pos[1] < y + @RaD)
```
Fixed by #4454 apparently.
